### PR TITLE
22 logged in user cant edit the personal data of any other account

### PR DESCRIPTION
**Pull request by daveking7**
Bug fix

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public UserService(@Qualifier("userRepository") UserRepository userRepository) {
 
   public User createUser(User newUser) {
     newUser.setToken(UUID.randomUUID().toString());
-    newUser.setStatus(UserStatus.OFFLINE);
+    newUser.setStatus(UserStatus.ONLINE);
```
Comment (serainaschraff) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:45]: are you sure that you want it as an enum and not as a boolean?

Comment (serainaschraff) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:45]: And I understand that it makes it easier if the status is online off the bat but technically i don't think it makes much sense. Yes the odds that I am able to create a user and then not able to log in with the newly created user are extremely slim but still I think creation per se should not be online.

File: src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ void createUser_validInputs_success() {
     assertEquals(testUser.getName(), createdUser.getName());
     assertEquals(testUser.getUsername(), createdUser.getUsername());
     assertNotNull(createdUser.getToken());
-    assertEquals(UserStatus.OFFLINE, createdUser.getStatus());
+    assertEquals(UserStatus.ONLINE, createdUser.getStatus());
```
Comment (serainaschraff) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java:55]: see Userservice.java

File: src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceIntegrationTest.java [hunk 3]
```diff
@@ -54,7 +54,7 @@ public void createUser_validInputs_success() {
     assertEquals(testUser.getName(), createdUser.getName());
     assertEquals(testUser.getUsername(), createdUser.getUsername());
     assertNotNull(createdUser.getToken());
-    assertEquals(UserStatus.OFFLINE, createdUser.getStatus());
+    assertEquals(UserStatus.ONLINE, createdUser.getStatus());
```
Comment (serainaschraff) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceIntegrationTest.java:57]: see Userservice.java

**Review by serainaschraff — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java (hunk 1 above)
Comment (daveking7) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java:45]: > are you sure that you want it as an enum and not as a boolean?

Yes pretty sure: We can extend the status in the future easily (to BUSY, TYPING, and so on)



> And I understand that it makes it easier if the status is online off the bat but technically i don't think it makes much sense. Yes the odds that I am able to create a user and then not able to log in with the newly created user are extremely slim but still I think creation per se should not be online.

You are right, the frontend shall make another request for loggin in after registration.

**Comment by daveking7** [comment]
If ok now, you can merge it

**Review by serainaschraff — APPROVED** [review]

**Pull request merged**